### PR TITLE
docs: add assurances in failure cases

### LINKED ISSUE #1001: Document how cilium agent behaves when it's dependencies are missing
What happens if kube-api/ kv-store goes away?
Can the cilium daemon survive a restart/outage of either of these services?
How long can the kv-store/kube-api be down before cilium has a problem?
What will happen to already running container?
What happens when creating a new container while kube-api/ kv-store is down?

### PR DESCRIPTION
**Pull request by aanm**
Fixes #1001 

Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

File: Documentation/concepts.rst [hunk 1]
```diff
@@ -143,6 +143,36 @@ The Key-Value (KV) Store is used for the following state:
...
+* Normal networking operations will continue;
+
+* If policy enforcement is enabled, the existing endpoints will still have their
+policy enforced but you will lose the ability to add additional containers that
+belong to security identities which are unknown on the node;
+
+* If services are enabled, you will lose the ability to add additional services
+ / loadbalancers;
+
+* When the connectivity is restored with KV-Store, cilium can take up to 5
```
Comment (ianvernon) [inline Documentation/concepts.rst]: ... restored to the KV-Store ...

File: Documentation/concepts.rst [hunk 2]
```diff
@@ -143,6 +143,36 @@ The Key-Value (KV) Store is used for the following state:
...
+minutes to re-sync the out-of-sync state with the KV-Store.
+
+Cilium will be kept running even in out-of-sync with the KV-Store.
```
Comment (ianvernon) [inline Documentation/concepts.rst]: Cilium will keep running even if it is out-of-sync with the KV-Store.

File: Documentation/concepts.rst [hunk 3]
```diff
@@ -143,6 +143,36 @@ The Key-Value (KV) Store is used for the following state:
...
+
+If cilium crashes / removed, it is guaranteed that:
```
Comment (ianvernon) [inline Documentation/concepts.rst]: What does "removed" mean in this context?

Comment (ianvernon) [reply Documentation/concepts.rst]: ... the following are guaranteed:

File: Documentation/concepts.rst [hunk 4]
```diff
@@ -143,6 +143,36 @@ The Key-Value (KV) Store is used for the following state:
...
+
+* When running cilium as a DaemonSet / container, with the specification files
```
Comment (ianvernon) [inline Documentation/concepts.rst]: What are these specification files ? Provide some sort of link here for clarity.

File: Documentation/concepts.rst [hunk 5]
```diff
@@ -143,6 +143,36 @@ The Key-Value (KV) Store is used for the following state:
...
+provided in the documentation, the endpoints / containers already running will
```
Comment (ianvernon) [inline Documentation/concepts.rst]: the endpoints / containers which are already running will not lose any connectivity, and they will keep running with the policy loaded before Cilium stopped unexpectedly.

File: Documentation/concepts.rst [hunk 6]
```diff
@@ -143,6 +143,36 @@ The Key-Value (KV) Store is used for the following state:
...
+not lose any connectivity and they will kept running with the policy loaded
+before cilium stopped unexpectedly.
+
+* When running cilium on a different way, just make sure the bpf fs is mounted
```
Comment (ianvernon) [inline Documentation/concepts.rst]: cilium --> Cilium

File: Documentation/concepts.rst [hunk 7]
```diff
@@ -143,6 +143,36 @@ The Key-Value (KV) Store is used for the following state:
 To simplify things in a larger deployment, the key-value store can be the same
 one used by the container orchestrater (e.g., Kubernetes using etcd).
 
+Assurances
+==========
+
+If cilium loses connectivity with the KV-Store, it guarantees that:
```
Comment (ianvernon) [inline Documentation/concepts.rst]: cilium --> Cilium

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: Documentation/concepts.rst (hunk 7 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

File: Documentation/concepts.rst (hunk 1 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

File: Documentation/concepts.rst (hunk 2 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

File: Documentation/concepts.rst (hunk 3 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

File: Documentation/concepts.rst (hunk 4 above)
Comment (aanm) [reply Documentation/concepts.rst]: If I want to refer a different file is it like this?

File: Documentation/concepts.rst (hunk 5 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

File: Documentation/concepts.rst (hunk 6 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

**Review by aalemayhu — CHANGES_REQUESTED** [review]
Looks good but would make a couple of changes so it looks better on PDF and HTML. The first underscore in the ref is preventing the links from being rendered, also you are missing some spacing for the paragraphs otherwise it becomes separate lines. Could you amend your commit with the below diff?

```diff
diff --git a/Documentation/concepts.rst b/Documentation/concepts.rst
index dab43063b951..006921ee6e33 100644
--- a/Documentation/concepts.rst
+++ b/Documentation/concepts.rst
@@ -150,15 +150,15 @@ If Cilium loses connectivity with the KV-Store, it guarantees that:
 
 * Normal networking operations will continue;
 
-* If policy enforcement is enabled, the existing endpoints will still have their
-policy enforced but you will lose the ability to add additional containers that
-belong to security identities which are unknown on the node;
+* If policy enforcement is enabled, the existing endpoints will still have
+  their policy enforced but you will lose the ability to add additional
+  containers that belong to security identities which are unknown on the node;
 
 * If services are enabled, you will lose the ability to add additional services
- / loadbalancers;
+  / loadbalancers;
 
 * When the connectivity is restored to the KV-Store, cilium can take up to 5
-minutes to re-sync the out-of-sync state with the KV-Store.
+  minutes to re-sync the out-of-sync state with the KV-Store.
 
 Cilium will keep running even if it is out-of-sync with the KV-Store.
 
@@ -166,12 +166,12 @@ If cilium crashes / or the DaemonSet is accidentally delete, the following are
 guaranteed:
 
 * When running cilium as a DaemonSet / container, with the specification files
-provided in the documentation :ref:`_admin_install_daemonset`, the endpoints /
-containers which are already running will not lose any connectivity, and they
-will keep running with the policy loaded before Cilium stopped unexpectedly.
+  provided in the documentation :ref:`admin_install_daemonset`, the endpoints /
+  containers which are already running will not lose any connectivity, and they
+  will keep running with the policy loaded before Cilium stopped unexpectedly.
 
 * When running Cilium on a different way, just make sure the bpf fs is mounted
-:ref:`_admin_mount_bpffs`.
+  :ref:`admin_mount_bpffs`.
 
 
 .. _labels:
```

**Review by aalemayhu — APPROVED** [review]

File: Documentation/concepts.rst [hunk 8]
```diff
@@ -143,6 +143,37 @@ The Key-Value (KV) Store is used for the following state:
...
+
+* If services are enabled, you will lose the ability to add additional services
+  / loadbalancers;
+
+* When the connectivity is restored to the KV-Store, cilium can take up to 5
+  minutes to re-sync the out-of-sync state with the KV-Store.
+
+Cilium will keep running even if it is out-of-sync with the KV-Store.
+
+If cilium crashes / or the DaemonSet is accidentally delete, the following are
```
Comment (ianvernon) [inline Documentation/concepts.rst]: delete --> deleted

File: Documentation/concepts.rst [hunk 9]
```diff
@@ -143,6 +143,37 @@ The Key-Value (KV) Store is used for the following state:
...
+* Normal networking operations will continue;
+
+* If policy enforcement is enabled, the existing endpoints will still have
+  their policy enforced but you will lose the ability to add additional
+  containers that belong to security identities which are unknown on the node;
+
+* If services are enabled, you will lose the ability to add additional services
+  / loadbalancers;
+
+* When the connectivity is restored to the KV-Store, cilium can take up to 5
```
Comment (ianvernon) [inline Documentation/concepts.rst]: Cilium

File: Documentation/concepts.rst (hunk 8 above)
Comment (ianvernon) [inline Documentation/concepts.rst]: cilium --> Cilium

File: Documentation/concepts.rst [hunk 10]
```diff
@@ -143,6 +143,37 @@ The Key-Value (KV) Store is used for the following state:
...
+  minutes to re-sync the out-of-sync state with the KV-Store.
+
+Cilium will keep running even if it is out-of-sync with the KV-Store.
+
+If cilium crashes / or the DaemonSet is accidentally delete, the following are
+guaranteed:
+
+* When running cilium as a DaemonSet / container, with the specification files
```
Comment (ianvernon) [inline Documentation/concepts.rst]: cilium --> Cilium

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: Documentation/concepts.rst (hunk 9 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

File: Documentation/concepts.rst (hunk 8 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

Comment (aanm) [reply Documentation/concepts.rst]: Fixed

File: Documentation/concepts.rst (hunk 10 above)
Comment (aanm) [reply Documentation/concepts.rst]: Fixed

**Review by ianvernon — APPROVED** [review]
LGTM!

**Review by nebril — APPROVED** [review]

**Review by aalemayhu — COMMENTED** [review]
Thanks for addressing comments, LGTM.

**Pull request merged**